preliz/distributions/laplace.py

```diff
@@ -0,0 +1,171 @@
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=arguments-differ
+import numpy as np
+import numba as nb
+
+from ..internal.distribution_helper import all_not_none, eps
+
+from .distributions import Continuous
+
+
+class Laplace(Continuous):
+    r"""
+    Laplace distribution.
+
+    The pdf of this distribution is
+
+    .. math::
+
+       f(x \mid \mu, b) =
+           \frac{1}{2b} \exp \left\{ - \frac{|x - \mu|}{b} \right\}
+
+    .. plot::
+        :context: close-figs
+
+        import arviz as az
+        from preliz import Laplace
+        az.style.use('arviz-white')
+        mus = [0., 0., 0., -5.]
+        bs = [1., 2., 4., 4.]
+        for mu, b in zip(mus, bs):
+            Laplace(mu, b).plot_pdf(support=(-10,10))
+
+    ========  ========================
+    Support   :math:`x \in \mathbb{R}`
+    Mean      :math:`\mu`
+    Variance  :math:`2 b^2`
+    ========  ========================
+
+    Parameters
+    ----------
+    mu : float
+        Location parameter.
+    b : float
+        Scale parameter (b > 0).
+    """
+
+    def __init__(self, mu=None, b=None):
+        super().__init__()
+        self.support = (-np.inf, np.inf)
+        self._parametrization(mu, b)
+
+    def _parametrization(self, mu=None, b=None):
+        self.mu = mu
+        self.b = b
+        self.params = (self.mu, self.b)
+        self.param_names = ("mu", "b")
+        self.params_support = ((-np.inf, np.inf), (eps, np.inf))
+        if all_not_none(mu, b):
+            self._update(mu, b)
+
+    def _update(self, mu, b):
+        self.mu = np.float64(mu)
+        self.b = np.float64(b)
+        self.params = (self.mu, self.b)
+        self.is_frozen = True
+
+    def pdf(self, x):
+        """
+        Compute the probability density function (PDF) at a given point x.
+        """
+        x = np.asarray(x)
+        return nb_pdf(x, self.mu, self.b)
+
+    def cdf(self, x):
+        """
+        Compute the cumulative distribution function (CDF) at a given point x.
+        """
+        x = np.asarray(x)
+        return nb_cdf(x, self.mu, self.b)
+
+    def ppf(self, q):
+        """
+        Compute the percent point function (PPF) at a given probability q.
+        """
+        q = np.asarray(q)
+        return nb_ppf(q, self.mu, self.b)
+
+    def logpdf(self, x):
+        """
+        Compute the log probability density function (log PDF) at a given point x.
+        """
+        return np.log(nb_pdf(x, self.mu, self.b))
```

Comment:
This should be the other way around, write an explicit expression for logpdf, and exponentiate that to compute the pdf.

Reply:
Noted, will keep that in mind....